src/homepage/howTo/custom-listeners.html

```diff
@@ -0,0 +1,217 @@
+                        which will listen events fired after emails are appended to users mailboxes,
+                        then for each email added event, determine the size of added email by getting size information from the event.
+                        If the size of the email is greater than or equals 1 MB, then BigMessageListener will add a "BIG_MESSAGE" flag to that email
+                    </p>
+
+                    <header class="major">
+                        <h3><b>Starting James with BigMessageListener</b></h3>
+                    </header>
+
+                    <p>
+                        When you write a custom Listener, you have to compile it and place the compiled jar package inside "extensions-jars" directory
+                        to make James load your Listener when it starts up. But BigMessageListener is already available in James products.
+                        So we can use it directly without that extra step.
+                    </p>
+
+                    <p>We will take James cassandra product for the example</p>
+
+                    <p>First, get template cassandra product configuration:</p>
+                    <pre><code>
+    $ git clone https://github.com/apache/james-project
+    $ cp -rf james-project/dockerfiles/run/guice/cassandra/destination/conf conf
+    $ mv keystore conf/keystore
+                    </code></pre>
+
+                    <p>
+                        Second, modify listener.xml configuration file in conf/ directory to add BigMessageListener
+                        by specifying its full class name
+                    </p>
+                    <pre><code>
+    &lt;listeners&gt;
+        ...
+        &lt;listener&gt;
+            &lt;class&gt;org.apache.james.examples.custom.listeners.BigMessageListener&lt;/class&gt;
+        &lt;/listener&gt;
+        ...
+    &lt;/listeners&gt;
+                    </code></pre>
+
+                    <p>Finally, starting a James Server by docker compose</p>
+                    <p>Getting James docker-compose.yml</p>
+                    <pre><code>
+    $ wget https://raw.githubusercontent.com/apache/james-project/master/dockerfiles/run/docker-compose.yml
+                    </code></pre>
+                    <p>Add the following volumes for james service:</p>
+                    <pre><code>
+    volumes:
+      - $PWD/conf:/root/conf/
+                    </code></pre>
+
+                    <p>
+                        If you are using your custom listeners, place the jar containing your listeners into "extensions-jars" directory
+                        and add a volume for it
+                    </p>
+                    <pre><code>
+    volumes:
+      - $PWD/extensions-jars:/root/extensions-jars/
+                    </code></pre>
+                    <p>
+                        Just like that, with a very simple preparation, you can make James use your custom listeners to serve your purpose.
```

Comment:
you lack the description of the test: send a mail bigger than 1MB and open the message in your mailer to see the flag (are custom flags visible in any way?)

Reply:
okay!

Reply:
I don't see anything about that, do I miss something?

Reply:
It takes me the morning to setting up James with the right configuration and recording necessary steps. So it arrives abit late. Please checkout 62d84cf85